sway/border.c

```diff
@@ -199,44 +313,85 @@ void update_view_border(swayc_t *view) {
 		}
 	}
 
-	switch (view->border_type) {
-	case B_NONE:
-		break;
-	case B_PIXEL:
-		cr = create_border_buffer(view, view->border_geometry, &surface);
-		if (!cr) {
-			break;
+	// for tabbed/stacked layouts the focused view has to draw all the
+	// titlebars of the hidden views.
+	swayc_t *p = swayc_tabbed_stacked_parent(view);
+	if (p && view->parent->focused == view) {
```

Comment:
You could make a bit of a performance improvement by checking view->parent->focused first and only get p only if you actually need it.

Downside: Makes the ifs a bit uglier.